relay-server/src/services/processor.rs

```diff
@@ -571,26 +571,17 @@ struct ProcessEnvelopeState<'a, Group> {
     /// extracted.
     event: Annotated<Event>,
 
-    /// Track whether transaction metrics were already extracted.
-    event_metrics_extracted: bool,
-
-    /// Track whether spans and span metrics were already extracted.
+    /// Item headers of the extracted event.
     ///
-    /// Only applies to envelopes with a transaction item.
-    spans_extracted: bool,
+    /// `None` if the envelope does not contain an event item.
```

Comment:
Should we just use a default here?

Reply:
`ItemHeaders` does not implement `Default` atm, it would require a default `ItemType` which sounds dangerous. In the end I think we should remove this from the state (see other comment).

Reply:
Should we use something like item headers instead which can represent the "default" case and also can be constructed from `ItemHeaders` as well as be turned into `Option<ItemHeaders>` again?